framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -315,7 +394,23 @@ public boolean isSubtype(AnnotationMirror rhs, AnnotationMirror lhs) {
                     }
                 }
                 return same;
+            } else if (AnnotationUtils.areSameByClass(lhs, IntRange.class)
+                    && AnnotationUtils.areSameByClass(rhs, IntVal.class)) {
+                List<Long> rhsValues =
+                        AnnotationUtils.getElementValueArray(rhs, "value", Long.class, true);
+                Range lhsRange = getIntRange(lhs);
+                long rhsMinVal = Collections.min(new ArrayList<>(rhsValues));
+                long rhsMaxVal = Collections.max(new ArrayList<>(rhsValues));
+                if (rhsMinVal >= lhsRange.from && rhsMaxVal <= lhsRange.to) {
```

Comment:
This is much simpler as `return rhsMinVal >= lhsRange.from && rhsMaxVal <= lhsRange.to`.